src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java

```diff
@@ -76,9 +89,178 @@ public MakeReservationFromDeck(int connectionHashCode, UserMessage userMessage,
+                    new TokensDataResponse(
+                        card.getCost(TokenType.RUBY), 
+                        card.getCost(TokenType.EMERALD), 
+                        card.getCost(TokenType.SAPPHIRE), 
+                        card.getCost(TokenType.DIAMOND), 
+                        card.getCost(TokenType.ONYX)
+                    )
+                )
+            );
+
+            ArrayList<User> players = room.getAllUsers();
+            ServerMessage serverMessage = new ServerMessage(
+                userMessage.getContextId(), 
+                ServerMessageType.MAKE_RESERVATION_FROM_DECK_ANNOUNCEMENT, 
+                Result.OK, 
+                responseData
+            );
+
+            for(User player : players){
+                messenger.addMessageToSend(player.getConnectionHashCode(), serverMessage);        
+            }
+
+        }catch (Exception e) {
+            ErrorResponse errorResponse = new ErrorResponse(Result.FAILURE, e.getMessage(), ServerMessageType.MAKE_RESERVATION_FROM_DECK_RESPONSE, userMessage.getContextId().toString());
+            messenger.addMessageToSend(connectionHashCode, errorResponse);
+        }
+    }
+
+    private void validateData(DataDTO dataDTO, Database database) throws Exception{
+        User user = database.getUser(dataDTO.userUuid);
+
+        //Check if user exists
+        if(user==null){
+            throw new UserDoesntExistException("There is no such user in the database");
+        }
+
+        Room room = database.getRoomWithUser(user.getUuid());
+
+        //Check if user is in any room
+        if(room==null){
+            throw new RoomDoesntExistException("User does not belong to any room");
+        }
+
+        Game game = room.getGame();
+        
+        //check if the room is in the game
+        if(game==null){
+             throw new RoomInGameException("Room is not in a game state");
+        }
+
+        //Check if it is user's turn
+        if(game.getCurrentPlayer()!=user){
+             throw new UserTurnException("It is not user's turn");
+        }
+        
+        //Check reservation count
+        if(user.getReservationCount()>=3){
+            throw new UserReservationException("User has to many reserved cards in hand");
+        }
+
+        //Check throughout game reservation count
+        if(user.getThroughoutGameReservationCount()>=5){
+            throw new UserReservationException("User reached the limit of reserved cards per game");
+        }
+
+        //Check if user has not maxed tokens
+        if(user.getTokenCount()>=10){
+            throw new TokenCountException("Reached max token count on hand");
+        }
+
+        //Check if if gold tokens are available
+        if(game.getTokenCount(TokenType.GOLD_JOKER)<=0){
+            throw new TokenCountException("There is not enough golden joker on table");
+        }
+
     }
     
-}
\ No newline at end of file
+}
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
I don't let those blank lines of code to be approved. If you want to get more lines of code, to be higher in the ranking you should add java docs to every class, method and field you've added. Free lines man.